src/React/Promise/Deferred.php

```diff
@@ -96,6 +122,35 @@ public function resolver()
         return $this->resolver;
     }
 
+    public function cancel()
+    {
+        if (null === $this->canceller || null !== $this->completed) {
+            return;
+        }
+
+        $canceller = $this->canceller;
+        $this->canceller = null;
+
```

Comment:
Alternatively, you could throw in the constructor, similar to [here](https://github.com/reactphp/promise/blob/1.0/src/React/Promise/Deferred.php#L30-L32).